tests/tests_geomstats/test_hypersphere.py

```diff
@@ -1,911 +1,701 @@
+        with expected:
+            space.tangent_extrinsic_to_spherical(
+                tangent_vec, base_point, base_point_spherical
+            )
 
-        self.assertAllClose(result, expected)
+    @geomstats.tests.np_autograd_and_torch_only
+    def test_riemannian_normal_frechet_mean(self, dim):
+        space = self.space(dim)
+        mean = space.random_uniform()
+        precision = gs.eye(space.dim) * 10
+        sample = space.random_riemannian_normal(mean, precision, 10000)
+        estimator = FrechetMean(space.metric, method="adaptive")
+        estimator.fit(sample)
+        estimate = estimator.estimate_
+        self.assertAllClose(estimate, mean, atol=1e-2)
 
-    def test_geodesic_and_belongs(self):
-        n_geodesic_points = 10
-        initial_point = self.space.random_uniform(2)
-        vector = gs.array([[2.0, 0.0, -1.0, -2.0, 1.0]] * 2)
-        initial_tangent_vec = self.space.to_tangent(
-            vector=vector, base_point=initial_point
-        )
-        geodesic = self.metric.geodesic(
-            initial_point=initial_point, initial_tangent_vec=initial_tangent_vec
-        )
-        t = gs.linspace(start=0.0, stop=1.0, num=n_geodesic_points)
-        points = geodesic(t)
-        result = gs.stack([self.space.belongs(pt) for pt in points])
+    @geomstats.tests.np_autograd_and_torch_only
+    def test_riemannian_normal_and_belongs(self, dim, n_points):
+        space = self.space(dim)
+        mean = space.random_uniform()
+        cov = gs.eye(dim)
+        sample = space.random_riemannian_normal(mean, cov, n_points)
+        result = space.belongs(sample)
         self.assertTrue(gs.all(result))
 
-        initial_point = initial_point[0]
-        initial_tangent_vec = initial_tangent_vec[0]
-        geodesic = self.metric.geodesic(
-            initial_point=initial_point, initial_tangent_vec=initial_tangent_vec
-        )
-        points = geodesic(t)
-        result = self.space.belongs(points)
-        expected = gs.array(n_geodesic_points * [True])
-        self.assertAllClose(expected, result)
-
-    def test_geodesic_end_point(self):
-        n_geodesic_points = 10
-        initial_point = self.space.random_uniform(4)
-        geodesic = self.metric.geodesic(
-            initial_point=initial_point[:2], end_point=initial_point[2:]
-        )
-        t = gs.linspace(start=0.0, stop=1.0, num=n_geodesic_points)
-        points = geodesic(t)
-        result = points[:, -1]
-        expected = initial_point[2:]
-        self.assertAllClose(expected, result)
-
-    def test_inner_product(self):
-        tangent_vec_a = gs.array([1.0, 0.0, 0.0, 0.0, 0.0])
-        tangent_vec_b = gs.array([0.0, 1.0, 0.0, 0.0, 0.0])
-        base_point = gs.array([0.0, 0.0, 0.0, 0.0, 1.0])
-        result = self.metric.inner_product(tangent_vec_a, tangent_vec_b, base_point)
-        expected = 0.0
-
-        self.assertAllClose(expected, result)
-
-    def test_inner_product_vectorization_single_samples(self):
```

Comment:
missing

Reply:
I have already  made it in part of inner_product.  other tests are not informative. because they are checking for shapes. which will be added in improvos